data-prepper-core/src/main/java/com/amazon/dataprepper/ContextManager.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.core.env.SimpleCommandLinePropertySource;
+
+/**
+ * @since 1.3
+ * <p>Creates Spring {@link org.springframework.context.ApplicationContext} hierarchy for Dependency Injection with limited visibility.</p>
+ * <p>
+ *     Application Context Hierarchy
+ *     <pre>
+ *         Public Application Context<br>
+ *         ├─ Core Application Context<br>
+ *         ├─ Shared Plugin Application Context<br>
+ *             ├─ Plugin Isolated Application Context<br>
+ *     </pre>
+ * </p>
+ */
+class ContextManager {
+    private static final Logger LOG = LoggerFactory.getLogger(ContextManager.class);
+
+    private final GenericApplicationContext publicApplicationContext;
+    private final AnnotationConfigApplicationContext coreApplicationContext;
+
+    /**
+     * @since 1.3
+     * @param args Application command line arguments
+     * 
+     * @see DataPrepperExecute#main(String...) 
+     */
+    public ContextManager(final String ... args) {
+        LOG.trace("Reading args");
+        final SimpleCommandLinePropertySource commandLinePropertySource = new SimpleCommandLinePropertySource(args);
+
+        publicApplicationContext = new GenericApplicationContext();
```

Comment:
This variable can be scoped to the constructor and does not need to be a class variable.